### PR TITLE
gh-110673: test_pty raises on short write

### PR DESCRIPTION
**Pull request by vstinner**
Add write_all() helper function to test_pty to raise an exception on short write: if os.writes() does not write all bytes. It should not happen for a PTY. Let's check if this assumption is correct.

* Issue: gh-110673

### CODE REVIEW

**Comment by vstinner** [comment]
I convert to a draft until I collect most (if not all) buildbot reports.

File: Lib/test/test_pty.py [hunk 1]
```diff
@@ -76,6 +76,17 @@ def expectedFailureIfStdinIsTTY(fun):
         pass
     return fun
 
+
+def write_all(fd, data):
+    while True:
```
Comment (serhiy-storchaka) [inline Lib/test/test_pty.py]: Why loop?

Comment (vstinner) [reply Lib/test/test_pty.py]: Oh. My first implementation (before I created this PR) actually implemented write_all(): retry os.write() until all bytes are written. But I was told that it must not happend, so I changed my PR to raise an exception instead.

I will remove the loop until buildbots complete, before merging this change.

At least, with this change, if the test hangs again, we will know for sure that no short write occurred.

**Comment by vstinner** [comment]
test_pty was run on a wide range of CI, GitHub Action and buildbots and it didn't fail. There were failures, but all unrelated.

**Pull request merged**